src/screens/Customers/CustomerMain.tsx

```diff
@@ -1,29 +1,130 @@
 import React, { useEffect, useState } from 'react';
-import { CustomerRecord } from '../../lib/airtable/interface';
-import { getAllCustomers } from '../../lib/airtable/request';
+import { CustomerRecord, SiteRecord } from '../../lib/airtable/interface';
 import { Link, RouteComponentProps } from 'react-router-dom';
+import { store } from '../../lib/redux/store';
+import { createStyles, FormControl, FormHelperText, MenuItem, Select, Theme, withStyles } from '@material-ui/core';
+import UserSearchBar from '../../components/UserSearchBar';
+import CustomerCard from '../../components/CustomerCard';
+import TrieTree from '../../lib/utils/TrieTree';
 
-function CustomerMain(props: RouteComponentProps) {
+const styles = (theme: Theme) =>
+  createStyles({
+    title: {
+      width: '153.9px',
+      height: '33.83px',
+      left: '25.6px',
+      top: '96px',
+      fontFamily: 'Helvetica Neue',
+      fontStyle: 'normal',
+      fontWeight: 500,
+      fontSize: '30px',
+      lineHeight: '115%',
+      color: '#828282',
+      flexGrow: 1,
+    },
+    headerWrapper: {
+      display: 'flex',
+      flexDirection: 'row',
+      width: '100%',
+      paddingTop: '86px',
+      paddingRight: '21px',
+      paddingLeft: '21px',
+    },
+    selectionHeader: {
+      display: 'flex',
+      flexDirection: 'column',
+      flexGrow: 1,
+    },
+    scrollDiv: {
+      maxHeight: '380px',
+      overflow: 'auto',
+      marginLeft: 'auto',
+      marginRight: 'auto',
+      width: '310.71px',
+    },
+  });
+
+interface UserProps {
+  classes: { title: string; headerWrapper: string; selectionHeader: string; scrollDiv: string };
+}
+
+function CustomerMain(props: RouteComponentProps & UserProps) {
   useEffect(() => {
     getCustomers();
   }, []);
 
+  const { classes } = props;
   const [customers, setCustomers] = useState<CustomerRecord[]>([]);
+  const [allCustomersTrie, setAllCustomersTrie] = useState<TrieTree<CustomerRecord>>(new TrieTree('name'));
+  const [fullCustomers, setFullCustomers] = useState<CustomerRecord[]>([]);
 
-  const getCustomers = async () => {
-    const customers: CustomerRecord[] = await getAllCustomers();
-    setCustomers(customers);
-  };
+  const getCustomers = () => {
+    const siteData = store.getState().siteData;
+    const sites: SiteRecord[] = siteData.sites;
+    let allCustomers: CustomerRecord[] = [];
+    for (let i = 0; i < sites.length; i++) {
+      let currCustomers: CustomerRecord[] = sites[i].customers;
+      allCustomers = allCustomers.concat(currCustomers)
+    }
```

Comment:
Doesn't this grab the customers from all the sites the user has access to? It's my understanding that we only want the customers of the current site the user has set.

Reply:
wait yeah that prob more apporpriate in regards to the app.